pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -53,12 +55,134 @@ func (p *singleNumaNodePolicy) canAdmitPodResult(hint *TopologyHint) lifecycle.P
 	}
 }
 
+// Get the narrowest hint aligned to one NUMA node
+// assumption: allResourcesHints are not empty, all hints have NUMAAffinity != nil,
+// all hints have exactly one NUMA node set in NUMANodeAffinity
+func (p *singleNumaNodePolicy) getHintMatch(allResourcesHints [][]TopologyHint) (bool, TopologyHint) {
```

Comment:
How is this function different than applying `iterateAllProviderTopologyHints()` over a filtered set of hints (i.e. those that only container a single numa node?